vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java

```diff
@@ -185,6 +182,7 @@ public void activate(BundleContext context, Config config) {
         log.info("Jackrabbit Filevault FS Package Registry initialized with home location {}", this.homeDir.getPath());
         this.scope = InstallationScope.valueOf(config.scope());
         this.securityConfig = new AbstractPackageRegistry.SecurityConfig(config.authIdsForHookExecution(), config.authIdsForRootInstallation());
+        loadPackageCache();
```

Comment:
I think doing file system accesses within the OSGI activation is dangerous - please see my comment https://github.com/apache/jackrabbit-filevault/pull/136#issuecomment-828974792 .

Reply:
The reason for not loading the cache here is because package operations are no frequent use case while the activate happens at any start of the system. This was designed for a Cloud-Service scenario with frequent starts & stops. Any expensive operation should be cut out of the startup procedure. The non-osgi initialization on the other hand is only required for initialization of an 'immutable' state of this is registry which would then only be read at system runtime (reused state) to validate dependencies for the mutable part being handled by the jcr registry. In case we have new scenarios we should make sure to not sacrifice those performance aspects.

Reply:
Clarification - installation of packages happens with an osgi startup at the moment as other mechanisms may write to repositorylike repo init but the accessible containers would only initialize the cache when there is package installation activity which is a rare scenario and is not supposed to delay startup

Reply:
Loading the package cache is crucial before running ExecutionPlanBuilder (look at https://github.com/apache/sling-org-apache-sling-jcr-packageinit/pull/7/files). But I will come up with a fix which keeps prevents reading the state from FileSystem in case the FSPackageRegistry is just referenced but no methods are called (as it is the case sometimes in https://github.com/apache/sling-org-apache-sling-jcr-packageinit/blob/master/src/main/java/org/apache/sling/jcr/packageinit/impl/ExecutionPlanRepoInitializer.java